tests/System.Text.Primitives.Tests/MemoryProtection/NativeMemory.Windows.cs

```diff
@@ -0,0 +1,277 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace System.Text.Primitives.Tests.MemoryProtection
+{
+    public static partial class NativeMemory
```

Comment:
It might be worth creating a subclass of OwnedMemory<T> that implements the protection semantics. Such subclass could be in System.Buffers.Experimental and then used in many tests.

Reply:
I agree. However I still have some questions as to how `OwnedMemory<T>` works and want to sit down with you and Ahson and others to get a better understanding of it. IMO I shouldn't let this block the merge, and I can create a new PR when that change is ready to come in.